docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+to create the entity. These `fromClientId` s are guaranteed to be sent by
+reticulum in a way that clients are unable to spoof.
 
-Note that `NetworkSchema` s are likely to change in the near future, as we are looking for ways to simplify the complexity that `serializeForStorage` and `deserializeForStorage` introduce.
+Entities that are created by calling `createNetworkedEntity` or receiving a
+`CreateMessage` are said to be `network instantiated`. `Network instantiated`
+entities may have many descendants. We do not say that the descendants are
+`network instantiated`.
 
 
-<a id="orgd0901c8"></a>
+#### UpdateMessage
+
+These tell a client to update an entity. Each `UpdateMessage` contains:
+
+- A `networkId`, which clients use to uniquely identify which entity the
+message refers to,
+- A `lastOwnerTime`, which tells clients when the sender of this message most
+recently witnessed ownership being transferred.
+- A `timestamp`, which indicates when a message was sent.
+- An `owner`, which indicates which `client` should have authority over
+updating this entity&rsquo;s state.
+
+Update messages also have the `data` needed to update an entity&rsquo;s state.
+An entity&rsquo;s state is simply the component data associated with this
+entity. Updates can be partial (updating only some components) or full
+(updating all components). Update messages also have two variants, depending
+on whether they are can be saved for long term storage in the database. This
+topic will be covered in another section.
+
+
+#### DeleteMessage
+
+These tell a client to `delete` an entity. Each `DeleteMessage` contains simply
+the `NetworkID` of the entity to be deleted. We distinguish between entities
+that have been `deleted` and those that are simply `removed`:
+
+- A `deleted` entity was explicitly deleted by a client. That is, someone
+pressed a button or took some action to delete it on purpose. Entities that
+have been `deleted` cannot be recreated.
+- A `removed` entity was removed incidentally. For example, it may have been
+removed when the `creator` disconnected from the room. If the `creator`
+reconnects and sends a `CreateMessage` with a matching `networkId`, it is
+acceptable to recreate the entity.
+
+
+#### ClientJoin
+
+These tell a client that a new client has connected. The next time the
+`networkSendSystem` runs, the receiving client will send the new client
+messages about entities it is the `creator` of, and update messages for
+entities it is the `owner` of.
+
+
+#### ClientLeave
+
+These tell a client that another client has disconnected. The next time the
+`networkReceiveSystem` runs, the receiving client will `remove` entities that
+the disconnected client was the `creator` of.
+
+
+#### CreatorChange
+
+These tell a client that the `creator` of an entity has been reassigned.
+Typically, this means that an entity has been `pinned` (or `unpinned`), and
+reticulum has assigned (or unassigned) itself as the entity&rsquo;s `creator`.

```

Comment:
Is reticulum itself the initial `creator` of an entity? Until this point my understanding was that the client that adds the network instantiated entity to the scene is the `creator` . If this is the case then maybe this is a sentence structure issue (”a client” instead of “itself”). If reticulum is the initial `creator` of a network instantiated entity then this should be explained when explaining the creator property.

Reply:
Same above.

Reply:
Added a TODO in #160 for now, thanks.